builtin/logical/aws/path_config_root.go

```diff
@@ -37,10 +41,16 @@ func pathConfigRoot() *framework.Path {
 func pathConfigRootWrite(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
 	region := data.Get("region").(string)
+	endpoint := data.Get("endpoint").(string)
+
+	if endpoint == "" {
+		endpoint = "none"
```

Comment:
You shouldn't use `"none"` as a placeholder value here. I'd just leave it as the empty string (which is the zero value for strings in golang) and compare against the empty string in client.go